ios/RCTMapboxGL/RCTMapboxGLManager.m

```diff
@@ -605,4 +605,143 @@ - (NSArray*)serializePacksArray:(NSArray<MGLOfflinePack*>*)packs
+                [geoJSONFeatures addObject:geoJSON];
+            }
+
+            resolve(geoJSONFeatures);
+        }
+    }];
+}
+
+- (NSDictionary*)geoJSONGeometryFromMGLFeature:(id <MGLFeature>)feature
+{
+    NSString *geometryType;
+
+    if ([feature isKindOfClass:[MGLShapeCollectionFeature class]]) {
+        geometryType = @"GeometryCollection";
+        MGLShapeCollectionFeature *shapeCollection = (MGLShapeCollectionFeature *) feature;
+        NSMutableArray *geometries = [[NSMutableArray alloc] init];
+        for (MGLShape <MGLFeature> *shape in shapeCollection.shapes) {
+            [geometries addObject:[self geoJSONGeometryFromMGLFeature:shape]];
+        }
+        return @{ @"type": geometryType,
+                  @"geometries": geometries };
+    }
+
+    NSMutableArray *coordinates = [[NSMutableArray alloc] init];
+
+    if ([feature isKindOfClass:[MGLPointFeature class]]) {
+        geometryType = @"Point";
+        coordinates = (NSMutableArray *) @[@(feature.coordinate.longitude), @(feature.coordinate.latitude)];
+    } else if ([feature isKindOfClass:[MGLPolylineFeature class]]) {
+        geometryType = @"LineString";
+        MGLPolylineFeature *polyline = (MGLPolylineFeature *)feature;
+        coordinates = [self getMGLPolylineCoordinates:polyline];
+    } else if ([feature isKindOfClass:[MGLPolygonFeature class]]) {
+        geometryType = @"Polygon";
+        MGLPolygonFeature *polygon = (MGLPolygonFeature *)feature;
+        coordinates = [self getMGLPolygonCoordinates:polygon];
+    } else if ([feature isKindOfClass:[MGLMultiPolylineFeature class]]) {
+        geometryType = @"MultiLineString";
+        MGLMultiPolylineFeature *multiPolyline = (MGLMultiPolylineFeature *)feature;
+        for (MGLPolylineFeature *polyline in multiPolyline.polylines) {
+            [coordinates addObject:[self getMGLPolylineCoordinates:polyline]];
+        }
+    } else if ([feature isKindOfClass:[MGLMultiPolygonFeature class]]) {
+        geometryType = @"MultiPolygon";
+        MGLMultiPolygonFeature *multiPolygon = (MGLMultiPolygonFeature *)feature;
+        for (MGLPolygonFeature *polygon in multiPolygon.polygons) {
+            [coordinates addObject:[self getMGLPolygonCoordinates:polygon]];
+        }
+    } else if ([feature isKindOfClass:[MGLMultiPointFeature class]]) {
+        // this is checked last since MGLPolyline and MGLPolygon inherit from MGLMultiPoint
+        geometryType = @"MultiPoint";
+        MGLMultiPointFeature *multiPoint = (MGLMultiPointFeature *)feature;
+        for (int index = 0; index < multiPoint.pointCount; index++) {
+            CLLocationCoordinate2D coord = multiPoint.coordinates[index];
+            [coordinates addObject:@[@(coord.longitude), @(coord.latitude)]];
+        }
+    }
+
+    return @{ @"type": geometryType,
+              @"coordinates": coordinates };
+}
+
+- (NSMutableArray *)getMGLPolylineCoordinates:(MGLPolylineFeature *)polyline
```

Comment:
An Objective-C method name should only begin with "get" if the method fetches a resource from the network or something of that nature.

A cleaner approach would be to add a category to MGLPolylineFeature with a method called `-coordinateArray`.